### ISSUE TITLE
Some character display misplaced in tmux

### ISSUE THREAD

**Issue by hanxi** [wez/wezterm#2567]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

20220927-071112-9be05951

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

If `windows -> ssh -> linux -> tmux -> nvim` steps, some character display misplaced, but `windows -> ssh -> linux -> nvim` steps display right.
Chinese characters and url like this:
```txt
call 回调 xxxxxxxxxxxxxxxxxxxxxcallback api:https://example.com/w#dxxxx
回调 xxxxxxxxxxxxxxxxxxxcallback api:https://example.com/w#dxxxxxx
```

In tmux:
![image](https://user-images.githubusercontent.com/1185757/193184310-49e79ebd-d990-44ff-badf-686530b28405.png)

Not in tmux:
![image](https://user-images.githubusercontent.com/1185757/193184339-d081327f-6814-448a-814b-58f2dc750e6e.png)

Another, in tig show git log also has this bug, use tmux and not use are same.
![image](https://user-images.githubusercontent.com/1185757/193182212-d55c447b-f498-43c0-9283-ff6532451685.png)

git log can get from <https://github.com/hanxi/dotfiles.git>

`TERM` is `screen-256color`

### Configuration

no config

### Logs

```txt
Debug Overlay
wezterm version: 20220927-071112-9be05951
OpenGL version: Quadro T1000 with Max-Q Design/PCIe/SSE2 4.5.0 NVIDIA 472.34
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
11:22:47.419 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
11:22:47.420 WARN wezterm_term::terminalstate > unhandled ResetMode(Unspecified(34))
11:22:47.420 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
11:22:47.420 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
11:22:47.420 WARN wezterm_term::terminalstate > unhandled ResetMode(Unspecified(34))
11:22:47.420 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
11:22:47.420 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
11:22:49.362 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
11:22:51.150 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
11:22:51.265 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
11:22:51.265 WARN wezterm_term::terminalstate > unhandled SetMode(Unspecified(34))
>
```

### Anything else?

`wezterm record` file: 
[wezterm-recording-hcKYrg.cast.txt](https://github.com/wez/wezterm/files/9680474/wezterm-recording-hcKYrg.cast.txt)

**Comment by wez** [comment]
This should be implemented now in `main` 13e57fa0b5df750dd68613f7558adf9964c38bf7

It typically takes about an hour before commits are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out. You can find the nightly downloads for your system in the [wezterm installation docs](https://wezfurlong.org/wezterm/installation.html).

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them. 

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by hanxi** [comment]
Thanks, it's ok now.
